libfreerdp/common/settings.c

```diff
@@ -1415,3 +1415,151 @@ BOOL freerdp_target_net_addresses_copy(rdpSettings* settings, char** addresses,
 
 	return TRUE;
 }
+
+BOOL proxy_parse_uri(rdpSettings* settings, const char* uri_in)
```

Comment:
Function should stay in `core/proxy.c`, it is not really something `settings` specific but a part of `proxy`